components/dada-parse/src/parser.rs

```diff
@@ -210,10 +210,8 @@ trait ParseList {
             // List items can always be separated by a newline...
             if !self.skipped_newline() {
                 // ...otherwise, they *may* require a separator
-                if comma_separated {
-                    if !self.eat_comma() {
-                        break;
-                    }
+                if comma_separated && !self.eat_comma() {
+                    break;
```

Comment:
oddly enough, I find this harder to read, I put the nesting there on purpose, but ... it's ok :)

Reply:
(my brain just has trouble parsing `&&` things...)

Reply:
https://rust-lang.github.io/rust-clippy/master/index.html#collapsible_if